### ISSUE TITLE
Tab screenshots will occasionally refresh to blank images

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#18034]
This issue is intermittent and not easy to replicate reliably. As best I can tell the steps are:

1. Open up several tabs
1. Background Firefox and allow it to remain idle for some time (duration is unknown)
1. Open Firefox

**Expected**: upon reopening Firefox and viewing tab tray, the previous tabs screenshots are visible

**Actual**: upon reopening and viewing tab tray, the previous tab screenshots are momentarily visible and then some of them will be set to a blank image

!IMG_6966.PNG|width=1125,height=2436!

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/FXIOS-8104)

**Comment by data-sync-user** [comment]
➤ Sophie Amin commented:

Copied from PR description: 
After investigating the issue I found that there was an additional call to takeScreenshot called in showLegacyTabTrayViewController that had no equivalent when the refactor was enabled, adding it made it so the most recent tab showed up more reliably.

As for issues after letting the app sit for awhile, I couldn't repro the older tabs issue with not loading the screenshot reliably without letting the app sit open for a long time, so it wasn't feasible to debug.

I could repro some loading issues when doing some navigation flows like Go to private mode from home page -> go back to a non private tab > open new tab > navigate away very quickly, but I think this might be because of the 500ms debounce time when taking screenshots after navigating which you can see in BVC func navigateInTab.

So, in short, issues probably still exist with this, but it should be better with this change in the tab tray refactor.

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Andrei Bodea Alina Moldovan I see this one is merged too, but still in QA needed. Can you check? Thanks!

**Comment by data-sync-user** [comment]
➤ Orla Mitchell commented:

For QA: I’m hoping the fix for this will impact both the regular tab tray and the tab tray refactor. This won’t fix screenshots that were blank prior to this update but for future tabs let me know if you see any blank images.

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Verified as fixed on v125 (40250), with iPhone 15 (17.0).

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Rechecked on v125 (40250), with iPhone 15 (17.0).

Orla Mitchell The issue seems to be fixed with tab tray refactor, but with regular tab tray, most of the tabs only display a favicon, not a screenshot, as in the video below. 

!original-550B2C7E-27A7-4364-946C-579DD470845E.mp4|width=444,height=960,alt="original-550B2C7E-27A7-4364-946C-579DD470845E.mp4"!

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

There is an increase of the incidence of this issue on v125 (40715), tested with iPad Air 3rd gen (17.3) and iPhone 15 (17.4.1).

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Verified as fixed on v9000 (41731), with iPhone 15 (17.4) and iPad Air 3rd gen (17.5).

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Verified as fixed on v126.1 (41973), with iPhone 15 (17.4) and iPad Air 3rd gen (17.5).
